packages/labs/src/components/timezone-select/timezoneSelect.tsx

```diff
@@ -0,0 +1,509 @@
+
+        const finalTargetRenderer = targetRenderer || this.defaultTargetRenderer;
+        return finalTargetRenderer({
+            date,
+            defaultDisplayValue: formatTimezone(finalDefaultValue, date, targetDisplayFormat),
+            defaultValue: finalDefaultValue,
+            disabled,
+            displayValue: formatTimezone(value, date, targetDisplayFormat),
+            placeholder: finalPlaceholder,
+            value,
+        });
+    }
+
+    private defaultTargetRenderer: ITimezoneSelectTargetRenderer = (targetProps) => {
+        const { targetClassName } = this.props;
+        const { value, displayValue, defaultDisplayValue, placeholder, disabled } = targetProps;
+        const isPlaceholder = !value;
+        const classes = classNames(Classes.TIMEZONE_SELECT_TARGET, targetClassName, {
+            [Classes.TIMEZONE_SELECT_TARGET_PLACEHOLDER]: isPlaceholder,
+        });
+
+        return (
+            <Button
+                className={classes}
+                text={displayValue || defaultDisplayValue || placeholder}
+                rightIconName="caret-down"
+                disabled={disabled}
+            />
+        );
+    }
+
+    private updateTimezones(date: Date): void {
+        const timezones = getTimezoneItems(date);
+        this.timezones = timezones;
+        this.timezoneToQueryCandidates = getTimezoneQueryCandidates(timezones, date);
+    }
+
+    private filterTimezones = (query: string, items: ITimezoneItem[]): ITimezoneItem[] => {
+        const normalizedQuery = normalizeText(query);
+        return items.filter((item) => {
+            const candidates = this.timezoneToQueryCandidates[item.timezone] || [normalizeText(item.timezone)];
+            return candidates.some((candidate) => isQueryMatch(normalizedQuery, candidate));
+        });
+    }
+
+    private renderItem = (itemProps: ISelectItemRendererProps<ITimezoneItem>) => {
+        const { item, isActive, handleClick } = itemProps;
+        const classes = classNames(CoreClasses.MENU_ITEM, CoreClasses.intentClass(), {
+            [CoreClasses.ACTIVE]: isActive,
+            [CoreClasses.INTENT_PRIMARY]: isActive,
+        });
+
+        return (
+            <MenuItem
+                key={item.key}
+                className={classes}
+                iconName={item.iconName}
+                text={item.text}
+                label={item.label}
+                onClick={handleClick}
+                shouldDismissPopover={false}
+            />
+        );
+    }
+
+    private handleItemSelect = (timezone: ITimezoneItem) => {
+        if (this.props.value === undefined) {
+            this.setState({ value: timezone.timezone });
+        }
+        Utils.safeInvoke(this.props.onChange, timezone.timezone);
+    }
+
+    private handleQueryChange = (query: string) => {
+        this.setState({ query });
+    }
+}
+
+function getTimezoneItems(date: Date): ITimezoneItem[] {
```

Comment:
let's pull these out to a `timezoneUtils` file in the same directory.

__OR__ a bunch of small utility files, one per function (some of these funcs are pretty complex)